### PR TITLE
Add "tray" package support

### PR DESCRIPTION
**Pull request by jmaupetit**
## Purpose

We've recently introduced the concept of distributable Arnold applications that we called "trays" [1]. Now we needed Arnold to support this new packaging format.

[1] https://github.com/openfun/fun-mooc/pull/9

## Proposal

- [x] make application paths configurable
- [x] move applications to the tray format
- [x] add tray support for the `apps` lookup plugin

During this work, I also had to:

- [x] move plugins to their standard location
- [x] fix various application warning

### CODE REVIEW

File: plugins/lookup/apps.py [hunk 1]
```diff
@@ -214,7 +214,9 @@ def run(self, terms, variables=None, **kwargs):
                         and settings_file in files
                     ):
                         with open(os.path.join(root, settings_file), "r") as stream:
-                            app["settings"].update(yaml.load(stream))
+                            app["settings"].update(
```
Comment (lunika) [inline plugins/lookup/apps.py]: In the commit title, the gitmoji used is not the good one. This one is about lint warning.

File: requirements.txt [hunk 2]
```diff
@@ -7,6 +7,7 @@ openshift==0.8.8
 passlib==1.7.1
 PyYAML==5.1
 netaddr==0.7.19
+requests==2.21.0
```
Comment (lunika) [inline requirements.txt:10]: In the commit title, the gitmoji used is not the good one. This one is about lint warning.

File: lookup_plugins/apps.py [hunk 3]
```diff
@@ -25,34 +19,32 @@
         applications' tree, e.g. for the "richie" application and the
         "apps" ``apps_path``:
 
-            apps/richie
+                apps/richie
```
Comment (lunika) [inline lookup_plugins/apps.py]: You can maybe remove 1 tab ?

File: lookup_plugins/apps.py [hunk 4]
```diff
@@ -25,34 +19,32 @@
...
-                        └── main.yml
+                    ├── all
+                    │   └── main.yml
+                    ├── settings.yml
+                    └── vault
+                        └── main.yml.j2
+
 
-        Analysing this tree with ``lookup_apps`` should return the
+        Analysing this tree with the ``apps`` lookup should return the
```
Comment (lunika) [inline lookup_plugins/apps.py:52]: in the datastructure expected, `environment_variables` is missing. This is my fault but you should add it in this commit.

File: tests/units/plugins/lookup/test_apps.py [hunk 5]
```diff
@@ -0,0 +1,345 @@
+"""
+Tests for the "apps" plugin lookup
+"""
+from pathlib import Path
```
Comment (lunika) [inline tests/units/plugins/lookup/test_apps.py]: This import is not used

File: tests/units/plugins/lookup/test_apps.py [hunk 6]
```diff
@@ -0,0 +1,345 @@
...
+
+from ansible.compat.tests import unittest
+from ansible.errors import AnsibleFilterError
```
Comment (lunika) [inline tests/units/plugins/lookup/test_apps.py]: This import seems not used.

File: requirements.txt (hunk 2 above)
Comment (jmaupetit) [reply requirements.txt:10]: I choose this emoji on purpose: to me, I removed kind of a linter warning. This is really subjective, but what would you recommend then?

File: plugins/lookup/apps.py (hunk 1 above)
Comment (jmaupetit) [reply plugins/lookup/apps.py]: Same remark here.

I choose this emoji on purpose: to me, I removed kind of a linter warning. This is really subjective, but what would you recommend then?

File: requirements.txt (hunk 2 above)
Comment (lunika) [reply requirements.txt:10]: Here you add a package, there is an emoji for that.

Comment (jmaupetit) [reply requirements.txt:10]: Nope the package is already installed, I manually fix an incompatibility (warning message).

Comment (lunika) [reply requirements.txt:10]: Then you pin it, there is also an emoji for that ;-)

**Review by lunika — COMMENTED** [review]
Everything look like ok for me. Maybe updating the documentation to explain the tray concept ? Maybe there is a reference to the old architecture and you have to update it ?

**Comment by jmaupetit** [comment]
> Maybe updating the documentation to explain the tray concept ?

I'll declare an issue, and write this documentation for a new PR if you don't mind.

> Maybe there is a reference to the old architecture and you have to update it ?

Nope there is not.

Deal?

**Review by lunika — APPROVED** [review]

**Comment by jmaupetit** [comment]
FTR I've declared #307

**Pull request merged**